src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -20,8 +20,10 @@ public static class ApiGenerator
     {
         static readonly string[] defaultWhitelistedNamespacePrefixes = new string[0];
 
-        public static string GeneratePublicApi(Assembly assembly, Type[] includeTypes = null, bool shouldIncludeAssemblyAttributes = true, string[] whitelistedNamespacePrefixes = null, string[] excludeAttributes = null)
+        public static string GeneratePublicApi(Assembly assembly, Type[]? includeTypes = null, bool shouldIncludeAssemblyAttributes = true, string[]? whitelistedNamespacePrefixes = null, string[]? excludeAttributes = null)
```

Comment:
if you decide to change the method signature, then let's not just place the `?` modifier on the set of parameters, but make new one class of options and move all the parameters there.

Reply:
This is not a change to the method signature. In an enabled nullability context, no other type is reasonable if you default to null. (I.e. they were already billable, you just couldn't see it before. The real change would be making them non-nullable.)

That being said, I already prefer the options type idea. Do you want to do something like https://github.com/ApiApprover/ApiApprover/issues/117#issuecomment-549068432? I'd prefer to do that 8n a separate PR.

An alternative would be to make PublicApiGenerator instantiable and put the same mutable instance properties on it rather than in a new options type.